htdocs/core/class/extrafields.class.php

```diff
@@ -959,7 +959,7 @@ public function showInputField($key, $value, $moreparam = '', $keysuffix = '', $
 
 		$label = $this->attributes[$extrafieldsobjectkey]['label'][$key];
 		$type = $this->attributes[$extrafieldsobjectkey]['type'][$key];
-		$size = $this->attributes[$extrafieldsobjectkey]['size'][$key];
+		$size = (float) $this->attributes[$extrafieldsobjectkey]['size'][$key];
```

Comment:
Size is a a string because it can be composed with 2 values, a first one for total digit and a second one for total nb of digit of decimal part.

Reply:
The $size is used in round on line 1002 and 1004 that creates a fatal error in PHP 8.x
The test is a compare with integer so the second part of the size is useless.

This create a fatal error on extrafields display just for a dumb test and so we can't create invoices !!!!

Reply:
> The $size is used in round on line 1002 and 1004 that creates a fatal error in PHP 8.x
> The test is a compare with integer so the second part of the size is useless.
> 
> This create a fatal error on extrafields display just for a dumb test and so we can't create invoices !!!!

I see. So i suggest to force the float only on line 1002 dand 1004, so when we need to use the value as a simple numeric, so we can keep the original value in memory.